### PR TITLE
create bcf issue based on existing work package

### PR DESCRIPTION
**Pull request by machisuji**
WP [#32493](https://community.openproject.com/projects/openproject/work_packages/32493/activity?query_id=2294)

### CODE REVIEW

**Comment by oliverguenther** [comment]
Frontend wise, this LGTM. The existing dev code can now create a new BCF :+1:

**Review by oliverguenther — APPROVED** [review]

File: modules/bim/app/services/bim/bcf/issues/create_service.rb [hunk 1]
```diff
@@ -34,20 +34,67 @@ class CreateService < ::BaseServices::Create
...
+
+      def work_package_id_from_links(links)
+        links
+          .take(1)
+          .map { |link| link.split("/").last.to_i }
+          .first
+      end
+
+      def remove_work_package_links!(links)
+        path = api_path_helper.work_package(42).sub("42", "")
```
Comment (ulferts) [inline modules/bim/app/services/bim/bcf/issues/create_service.rb]: As the path helper is simpler than one would expect the following also works:
```
path = api_path_helper.work_package("")
```

File: modules/bim/app/services/bim/bcf/issues/create_service.rb [hunk 2]
```diff
@@ -34,20 +34,67 @@ class CreateService < ::BaseServices::Create
...
-          wp_call
+          create_work_package params
+        end
+      end
+
+      def use_work_package(links:, params:)
+        work_package = WorkPackage.find_by(id: work_package_id_from_links(links))
+        return work_package_not_found_result if work_package.nil?
+
+        work_package.assign_attributes params
```
Comment (ulferts) [inline modules/bim/app/services/bim/bcf/issues/create_service.rb]: You should use the `WorkPackages::UpdateService` here. Doing so will e.g.:

* Check whether the user is allowed to edit/view the work package
* Check the attributes for validity.
* Return a service result.

**Review by ulferts — CHANGES_REQUESTED** [review]
Apart from what I wrote within the code, we also need to have a validation, probably in the issue contract, that prevents providing a work package that already has an issue assigned. Currently this is only prevented by a database index but leads to a 500.

File: modules/bim/app/services/bim/bcf/issues/create_service.rb (hunk 2 above)
Comment (machisuji) [reply modules/bim/app/services/bim/bcf/issues/create_service.rb]: That makes sense.

File: modules/bim/app/services/bim/bcf/issues/create_service.rb (hunk 1 above)
Comment (machisuji) [reply modules/bim/app/services/bim/bcf/issues/create_service.rb]: TIL!

**Comment by machisuji** [comment]
@ulferts I've added the missing validation.

**Review by ulferts — APPROVED** [review]
👍

**Pull request merged**